### PR TITLE
feat: add better profitable checks

### PR DESCRIPTION
**Pull request by TheRyanMiller**
Trying this again...
Improved tests for `test_profitable_harvest`.

- Add a check to ensure pricePerShare has increased
- Include vault balance for profit check rather than the strategy
- `user` instead of `gov`
- Add sender for harvest txns (makes it easier to test against live contracts)

### CODE REVIEW

File: tests/test_operation.py [hunk 1]
```diff
@@ -19,57 +20,61 @@ def test_operation(
...
-    accounts, token, vault, strategy, strategist, amount, RELATIVE_APPROX
+    accounts, token, vault, strategy, user, strategist, amount, RELATIVE_APPROX
 ):
     # Deposit to the vault
-    token.approve(vault.address, amount, {"from": accounts[0]})
-    vault.deposit(amount, {"from": accounts[0]})
-    strategy.harvest()
+    token.approve(vault.address, amount, {"from": user})
+    vault.deposit(amount, {"from": user})
+    strategy.harvest({"from": strategist})
```
Comment (steffenix) [inline tests/test_operation.py:33]: The object strategy has been created using the strategist account, you don't need to specify it again.

File: tests/test_operation.py [hunk 2]
```diff
@@ -19,57 +20,61 @@ def test_operation(
...
+    token.approve(vault.address, amount, {"from": user})
+    vault.deposit(amount, {"from": user})
     assert token.balanceOf(vault.address) == amount
 
-    # harvest
-    strategy.harvest()
-    assert pytest.approx(strategy.estimatedTotalAssets(), rel=RELATIVE_APPROX) == amount
+    # Harvest 1: Send funds through the strategy
+    strategy.harvest({"from": strategist})
+    assert strategy.estimatedTotalAssets() == amount
```
Comment (steffenix) [inline tests/test_operation.py:52]: You should keep the approximation, the base strategy is fairly simple but harvest might deposit into a contract that does some complex calculation and rounding, the test will fail.

File: tests/test_operation.py [hunk 3]
```diff
@@ -96,19 +101,20 @@ def test_sweep(gov, vault, strategy, token, amount, weth, weth_amout):
...
 
-    weth.transfer(strategy, weth_amout, {"from": gov})
+    before_balance = weth.balanceOf(gov)
+    weth.transfer(strategy, weth_amout, {"from": user})
     assert weth.address != strategy.want()
-    assert weth.balanceOf(gov) == 0
+    assert weth.balanceOf(user) == 0
     strategy.sweep(weth, {"from": gov})
-    assert weth.balanceOf(gov) == weth_amout
+    assert weth.balanceOf(gov) - before_balance == weth_amout
```
Comment (steffenix) [inline tests/test_operation.py:109]: assert weth.balanceOf(gov)  == before_balance + weth_amout

File: tests/test_operation.py (hunk 1 above)
Comment (TheRyanMiller) [reply tests/test_operation.py:33]: Reason I changed this one is to be more portable for testing against already-deployed strategies/vaults.

File: tests/test_operation.py (hunk 2 above)
Comment (TheRyanMiller) [reply tests/test_operation.py:52]: OK

File: tests/test_operation.py (hunk 3 above)
Comment (TheRyanMiller) [reply tests/test_operation.py:109]: same difference, but yours looks better I agree :)

File: yarn.lock [hunk 4]
```diff
@@ -2815,4 +2815,4 @@ yargs@^15.1.0:
 yocto-queue@^0.1.0:
   version "0.1.0"
   resolved "https://registry.yarnpkg.com/yocto-queue/-/yocto-queue-0.1.0.tgz#0294eb3dee05028d31ee1a5fa2c556a6aaf10a1b"
-  integrity sha512-rVksvsnNCdJ/ohGc6xgPwyN8eheCxsiLM8mxuE/t/mOVqJewPuO1miLpTHQiRgTKCLexL4MeAFVagts7HmNZ2Q==
+  integrity sha512-rVksvsnNCdJ/ohGc6xgPwyN8eheCxsiLM8mxuE/t/mOVqJewPuO1miLpTHQiRgTKCLexL4MeAFVagts7HmNZ2Q==
```
Comment (steffenix) [inline yarn.lock:2818]: Did you auto-format the lock file?

**Review by steffenix — APPROVED** [review]
Good thing to distinguish user and gov, that will prevent mistakes from being made.

🎉🥳👏

File: yarn.lock (hunk 4 above)
Comment (TheRyanMiller) [reply yarn.lock:2818]: Not certain exactly what caused this one.

**Pull request merged**